integration/docker_integration_test_executor.go

```diff
@@ -0,0 +1,129 @@
+// +build integration
+
+package main
+
+import (
+	"bytes"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+// IntegrationTestDockerExecRunner keeps the state of an instance of a docker runner
+type IntegrationTestDockerExecRunner struct {
+	// Runner is the ExecRunner to which all executions are forwarded in the end.
+	Runner        command.Command
+	Image         string
+	User          string
+	TestDir       []string
+	Mounts        map[string]string
+	Environment   map[string]string
+	Setup         []string
+	ContainerName string
+}
+
+// IntegrationTestDockerExecRunnerBundle is used to construct an instance of IntegrationTestDockerExecRunner
+type IntegrationTestDockerExecRunnerBundle struct {
+	Image       string
+	User        string
+	TestDir     []string
+	Mounts      map[string]string
+	Environment map[string]string
+	Setup       []string
+}
+
+func givenThisContainer(t *testing.T, bundle IntegrationTestDockerExecRunnerBundle) IntegrationTestDockerExecRunner {
+	runner := command.Command{}
+
+	// Generate a random container name so we can start a new one for each test method
+	// We don't rely on docker's random name generator for two reasons
+	// First, it is easier to save the name here compared to getting it from stdout
+	// Second, the common prefix allows batch stopping/deleting of containers if so desired
+	// The test code will not automatically delete containers as they might be useful for debugging
```

Comment:
Maybe a bit misleading. It will automatically delete containers, but with a delay. Or did I misunderstand this part?

Reply:
delete != stop